plugin/pkg/utils/miscellaneous.go

```diff
@@ -88,3 +91,35 @@ func ValidateZoneRemovalFromSeeds(oldSeedSpec, newSeedSpec *core.SeedSpec, seedN
 
 	return nil
 }
+
+// ValidateSeedNetworksUpdateWithShoots returns an error when seed networks are changed so that they are incompatible
+// with shoots scheduled on this seed.
+func ValidateSeedNetworksUpdateWithShoots(oldSeedSpec, newSeedSpec *core.SeedSpec, seedName string, shootLister gardencorev1beta1listers.ShootLister, kind string) error {
+	// for now, we only need to validate changes to the VPN network, the other seed networks are immutable.
+	if apiequality.Semantic.DeepEqual(oldSeedSpec.Networks.VPN, newSeedSpec.Networks.VPN) {
+		return nil
+	}
+
+	shoots, err := ListShootsUsingSeed(seedName, shootLister)
+	if err != nil {
+		return apierrors.NewInternalError(err)
+	}
+
+	for _, shoot := range shoots {
+		if len(cidrvalidation.ValidateNetworkDisjointedness(
+			nil,
+			shoot.Spec.Networking.Nodes,
+			shoot.Spec.Networking.Pods,
+			shoot.Spec.Networking.Services,
+			newSeedSpec.Networks.Nodes,
+			newSeedSpec.Networks.VPN,
+			newSeedSpec.Networks.Pods,
+			newSeedSpec.Networks.Services,
+			v1beta1helper.IsWorkerless(shoot),
+		)) > 0 {
+			return apierrors.NewForbidden(core.Resource(kind), seedName, fmt.Errorf("cannot update networks of %s %s as there are Shoots with overlapping networks scheduled to this Seed", kind, seedName))
+		}
+	}
```

Comment:
Why not add more context to the error message? The validation actually checks, which shoot has an overlap in what range, but we simply discard this knowledge and only return a generic message.

Reply:
Done. I hope this is what you had in mind.